yarn-project/simulator/src/client/view_data_oracle.ts

```diff
@@ -368,11 +368,11 @@ export class ViewDataOracle extends TypedOracle {
 
   // TODO(#11849): consider replacing this oracle with a pure Noir implementation of aes decryption.
   public override aes128Decrypt(ciphertext: Buffer, iv: Buffer, symKey: Buffer): Promise<Buffer> {
-    // Noir can't predict the amount of padding that gets trimmed,
-    // but it needs to know the length of the returned value.
-    // So we tell Noir that the length is the (predictable) length
-    // of the padded plaintext, we return that padded plaintext, and have Noir interpret the padding to do the trimming.
+    // In Noir-land, when this oracle is called, all the caller knows is the
+    // ciphertext length. Without performing decryption, the caller cannot know the
+    // true plaintext length. Therefore, we return a BoundedVec, which conveys the
+    // actual length of the plaintext.
```

Comment:
We're not creating the bounded vec here though

Reply:
Removed